jit/src/main/java/org/dcsa/conformance/standards/jit/model/PortCallServiceEventTypeCode.java

```diff
@@ -0,0 +1,76 @@
+package org.dcsa.conformance.standards.jit.model;
+
+import static org.dcsa.conformance.standards.jit.model.PortCallServiceType.*;
+
+import java.util.Arrays;
+import java.util.List;
+import lombok.Getter;
+import org.dcsa.conformance.core.UserFacingException;
+
+@Getter
+public enum PortCallServiceEventTypeCode {
+  STRT("Started"),
+  CMPL("Completed"),
+  ARRI("Arrived"),
+  DEPA("Departed");
+
+  private final String fullName;
+
+  PortCallServiceEventTypeCode(String fullName) {
+    this.fullName = fullName;
+  }
+
+  public static List<PortCallServiceType> getValidPortCallServiceTypes(
+      PortCallServiceEventTypeCode code) {
+    return switch (code) {
+      case STRT, CMPL ->
+          List.of(
+              CARGO_OPERATIONS,
+              PILOTAGE,
+              TOWAGE,
+              MOORING,
+              BUNKERING,
+              ANCHORAGE,
+              SLUDGE,
+              SEA_PASSAGE,
+              DISCHARGE_CARGO_OPERATIONS,
+              LOADING_CARGO_OPERATIONS,
+              LASHING,
+              ANCHORAGE_OPERATIONS,
+              SHORE_POWER);
+      case ARRI ->
+          List.of(
+              BERTH,
+              PILOT_BOARDING_PLACE,
+              ALL_FAST,
+              GANGWAY_DOWN_AND_SECURE,
+              VESSEL_READY_FOR_CARGO_OPERATIONS,
+              VESSEL_READY_TO_SAIL,
+              MOVES);
+      case DEPA -> List.of(SAFETY, BERTH);
+    };
```

Comment:
Nit: Since the result is always used with .contains, then in theory Set.of would be a better solution. Data size is probably too small to matter a lot, so feel free to dismiss.